msu/systems/registry/registry_mod_addon.nut

```diff
@@ -0,0 +1,63 @@
+::MSU.Class.RegistryModAddon <- class extends ::MSU.Class.SystemModAddon
```

Comment:
Overall, I am wondering if it is better to have a single `addURL` function exposed to the public API instead of separate functions for different websites e.g. `addGitHubURL` and `addNexusModsURL`. We may want to have another website in the future, and this adds functions for each website. With a single `addURL` function we can use the regex to automatically detect what domain the URL belongs to, and if it is github we set it as `__GitHubURL` and and respectively for other internal variables.

Reply:
I don't really wanna do that since it makes it harder to give a useful error msg, if we make a LinkSource enum or smtng and then pass those together that would be fine by me but idk if that's really much better...